### PR TITLE
draft steth pricer

### PR DESCRIPTION
**Pull request by chuddster**
# Task: STETH pricer

## High Level Description

Specific Changes

* Created a new pricer, STETHPricer
* The Pricer uses the stETH's getPooledEthByShares function to get the price of the stETH token in USD. [reference].
* The logic is similar to CompoundPricer, where it requires the USD price of the underlying to be committed beforehand.
* Unit tests for STETHPricer

### Code

- [ ] Unit test 100% coverage
- [ ] Does your code follow the naming and code documentation guidelines?

### Documentation

- [ ] Is your code up to date with the spec? 
- [ ] Have you added your tests to the testing doc?

### CODE REVIEW

File: contracts/interfaces/STETHInterface.sol [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.6.10;
+
+interface STETHInterface {
```
Comment (aleone) [inline contracts/interfaces/STETHInterface.sol:4]: We need to use the wstETH interface here, which doesn't include getPooledEthByShares.

File: contracts/mocks/MockSTETHToken.sol [hunk 2]
```diff
@@ -0,0 +1,21 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.6.10;
+
+import {ERC20Upgradeable} from "../packages/oz/upgradeability/ERC20Upgradeable.sol";
+
+contract MockSTETHToken is ERC20Upgradeable {
```
Comment (aleone) [inline contracts/mocks/MockSTETHToken.sol:6]: This is a mock for stETH, but we need a wstETH token, which has doesn't have getPooledEthByShares.

File: contracts/pricers/STETHPricer.sol [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+import {OracleInterface} from "../interfaces/OracleInterface.sol";
+import {OpynPricerInterface} from "../interfaces/OpynPricerInterface.sol";
+import {STETHInterface} from "../interfaces/STETHInterface.sol";
+import {ERC20Interface} from "../interfaces/ERC20Interface.sol";
+import {SafeMath} from "../packages/oz/SafeMath.sol";
+
+/**
+ * @notice A Pricer contract for a Yearn yToken
+ */
+contract STETHPricer is OpynPricerInterface {
```
Comment (aleone) [inline contracts/pricers/STETHPricer.sol:13]: I think we should probably call it WSTETHPricer as it is pricing wstETH collateral.

File: contracts/pricers/STETHPricer.sol [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        oracle.setExpiryPrice(address(stETH), _expiryTimestamp, stETHPrice);
+    }
+
+    /**
+     * @dev convert underlying price to stETH price with the stETH to underlying exchange rate
+     * @param _underlyingPrice price of 1 underlying token (ie 1e18 WETH) in USD, scaled by 1e8
+     * @return price of 1e8 stETH in USD, scaled by 1e8
+     */
+    function _underlyingPriceToSTETHPrice(uint256 _underlyingPrice) private view returns (uint256) {
+        uint256 pricePerShare = stETH.getPooledEthByShares(1 ether);
```
Comment (aleone) [inline contracts/pricers/STETHPricer.sol:73]: Here we can use `stEthPerToken()` from the wstETH contract

https://docs.lido.fi/contracts/wsteth

File: contracts/pricers/STETHPricer.sol [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    /// @notice underlying asset for this yToken
+    ERC20Interface public underlying;
+
+    /**
+     * @param _stETH stETH
+     * @param _underlying underlying asset for stETH
+     * @param _oracle Opyn Oracle contract address
+     */
+    constructor(
+        address _stETH,
```
Comment (aleone) [inline contracts/pricers/STETHPricer.sol:31]: Throughout, I think should change to wstETH to make it clear we are collateralizing with wstETH.

**Review by aleone — CHANGES_REQUESTED** [review]

File: test/unit-tests/stETHPricer.test.ts [hunk 6]
```diff
@@ -0,0 +1,126 @@
+import {
+  MockPricerInstance,
+  MockOracleInstance,
+  MockERC20Instance,
+  MockSTETHTokenInstance,
```
Comment (aleone) [inline test/unit-tests/stETHPricer.test.ts:5]: Tests need to be updated w new mocks, interface, functions, etc

**Comment by haythem96** [comment]
Closing this in favor for #440

**Pull request closed without merging**